pkg/kubelet/kubelet_pods.go

```diff
@@ -1726,6 +1697,46 @@ func getPhase(pod *v1.Pod, info []v1.ContainerStatus, podIsTerminal bool) v1.Pod
 	}
 }
 
+// handleRestartableInitContainers handles restartable initialization containers.
+func handleRestartableInitContainers(spec v1.PodSpec, info []v1.ContainerStatus, unknown, running, waiting int) (int, int, int, int) {
```

Comment:
code style: let's keep this inline and not as a separate function. All other loops are there and it just easier to understand what we do if it's all together

Reply:
> code style: let's keep this inline and not as a separate function. All other loops are there and it just easier to understand what we do if it's all together

This modification is due to your previous comment. I might not have fully understood the true intent behind the following statement.

> as implemented it is hard to reason whether the new behavior can be disabled by turning the feature gate off. Can you please make logic more compartmentalized so we can reason about it easier from that perspective.